src/modules/reporting/components/reporting-report-form/reporting-report-form.jsx

```diff
@@ -1,201 +1,231 @@
-            onClick={(e) => { validateIsMarketValid(p.validations, p.updateState, true) }}
-          >Yes
-          </button>
-        </li>
+
+    return (
+      <ul className={classNames(Styles.ReportingReportForm__fields, FormStyles.Form__fields)}>
         <li>
-          <button
-            className={classNames({ [`${FormStyles.active}`]: p.isMarketValid === false })}
-            onClick={(e) => { validateIsMarketValid(p.validations, p.updateState, false) }}
-          >No
-          </button>
-        </li>
-      </ul>
-    </li>
-    { p.isMarketValid && (p.market.type === BINARY || p.market.type === CATEGORICAL) &&
-      <li>
-        <label>
-          <span>Outcome</span>
-        </label>
-        <ul className={FormStyles['Form__radio-buttons--per-line']}>
-          { p.market.outcomes && p.market.outcomes.sort((a, b) => b.stake - a.stake).map(outcome => (
-            <li key={outcome.id}>
-              <button
-                className={classNames({ [`${FormStyles.active}`]: p.selectedOutcome === outcome.name })}
-                onClick={(e) => { validateOutcome(p.validations, p.updateState, outcome.name) }}
-              >{outcome.name} &nbsp;|&nbsp; {outcome.stake} REP
-              </button>
-            </li>
-          ))
-          }
-        </ul>
-      </li>
-    }
-    { p.isMarketValid && p.market.type === SCALAR &&
-      <li className={FormStyles['field--short']}>
-        <label>
-          <span htmlFor="sr__input--outcome-scalar">Outcome</span>
+          <label>
+            <span>Outcome</span>
+          </label>
           { p.validations.hasOwnProperty('selectedOutcome') && p.validations.selectedOutcome.length &&
             <span className={FormStyles.Form__error}>
-              { p.validations.selectedOutcome }
+              {InputErrorIcon}{ p.validations.selectedOutcome }
             </span>
           }
-        </label>
-        <input
-          id="sr__input--outcome-scalar"
-          type="number"
-          min={p.market.minValue}
-          max={p.market.maxValue}
-          placeholder="0"
-          value={p.selectedOutcome}
-          onChange={(e) => { validateNumber(p.validations, p.updateState, 'selectedOutcome', e.target.value, 'outcome', p.market.minValue, p.market.maxValue) }}
-        />
-      </li>
-    }
-    <li className={FormStyles['field--short']}>
-      <label>
-        <span htmlFor="sr__input--stake">Stake</span>
-        { p.validations.hasOwnProperty('stake') && p.validations.stake.length &&
-          <span className={FormStyles.Form__error}>
-            { p.validations.stake }
-          </span>
+        </li>
+        { (p.market.marketType === BINARY || p.market.marketType === CATEGORICAL) &&
+          <li>
+            <ul className={FormStyles['Form__radio-buttons--per-line']}>
+              { s.outcomes && s.outcomes.sort((a, b) => b.stake - a.stake).map(outcome => (
```

Comment:
This kind of sorting should really happen at the component state level so that it’s not re-sorting every time the component re-renders.

Reply:
I'll move to the component state level and out of the render.